imdclient/IMDClient.py

```diff
@@ -961,23 +953,41 @@ def __init__(self, n_atoms, imdsinfo):
             }
         else:
             self.energies = None
-        if imdsinfo.box:
+        if imdsinfo and imdsinfo.box:
             self.box = np.empty((3, 3), dtype=np.float32)
         else:
             self.box = None
-        if imdsinfo.positions:
+        if imdsinfo and imdsinfo.positions:
             self.positions = np.empty((n_atoms, 3), dtype=np.float32)
         else:
             self.positions = None
-        if imdsinfo.velocities:
+        if imdsinfo and imdsinfo.velocities:
             self.velocities = np.empty((n_atoms, 3), dtype=np.float32)
         else:
             self.velocities = None
-        if imdsinfo.forces:
+        if imdsinfo and imdsinfo.forces:
             self.forces = np.empty((n_atoms, 3), dtype=np.float32)
         else:
             self.forces = None
 
+    def copy(self):
+        """
+        Returns a deep copy of the IMDFrame
+        """
+        new_frame = IMDFrame(
```

Comment:
Did you try if [copy.deepcopy()](https://docs.python.org/3/library/copy.html#copy.deepcopy) would just do all the work for you? Then you wouldn't need the hack with `imsdinfo = None`.